src/Writer/BatchWriter.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Ddeboer\DataImport\Writer;
+
+use Ddeboer\DataImport\Writer;
+
+/**
+ * @author Markus Bachmann <markus.bachmann@bachi.biz>
+ */
+class BatchWriter implements Writer
+{
+    private $delegate;
+
+    private $size;
+
+    private $queue;
+
+    public function __construct(Writer $delegate, $size = 20)
+    {
+        $this->delegate = $delegate;
+        $this->size = $size;
+    }
+
+    public function prepare()
+    {
+        $this->delegate->prepare();
+
+        $this->queue = new \SplQueue();
```

Comment:
I use the `SplQueue` because of its function to delete the item after its iterated. But yes it uses more memory (except of HHVM).

Array: http://3v4l.org/FQHCd
SplQueue: http://3v4l.org/5pUns